test/tests/bootstrap/test_api20_os_bootstrap_parallel_local.py

```diff
@@ -0,0 +1,297 @@
+                                "hostname": "rackhdnode",
+                                "users": [{"name": "rackhduser",
+                                           "password": "RackHDRocks!",
+                                           "uid": 1010}]}}}
+                # OS specific payload requirements
+                if item['workflow'] == "Graph.InstallUbuntu":
+                    payload_data["options"]["defaults"]["baseUrl"] = "install/netboot/ubuntu-installer/amd64"
+                    payload_data["options"]["defaults"]["kargs"] = {"live-installer/net-image": rackhdhost +
+                                                                    item['path'] + "/ubuntu/install/filesystem.squashfs"}
+                if item['workflow'] == "Graph.InstallWindowsServer":
+                    payload_data["options"]["defaults"]["productkey"] = item['productkey']
+                    payload_data["options"]["defaults"]["smbUser"] = item['smbUser']
+                    payload_data["options"]["defaults"]["smbPassword"] = item['smbPassword']
+                    payload_data["options"]["defaults"]["smbRepo"] = "\\\\" + str(rackhdconfig['apiServerAddress']) + \
+                                                                     item['smbRepo']
+                    payload_data["options"]["defaults"]["username"] = "rackhduser"
+                    payload_data["options"]["defaults"]["password"] = "RackHDRocks!"
+                    payload_data["options"]["defaults"].pop('rootPassword', None)
+                    payload_data["options"]["defaults"].pop('users', None)
+                    payload_data["options"]["defaults"].pop('kvm', None)
+                    payload_data["options"]["defaults"].pop('version', None)
+
+                # run workflow
+                result = fit_common.rackhdapi('/api/2.0/nodes/' +
+                                              NODECATALOG[nodeindex] +
+                                              '/workflows?name=' + item['workflow'],
+                                              action='post', payload=payload_data)
+                if result['status'] == 201:
+                    # this saves the task and node IDs
+                    NODE_STATUS[NODECATALOG[nodeindex]] = \
+                        {"workflow": item['workflow'],
+                         "version": item['version'],
+                         "kvm": item['kvm'],
+                         "id": result['json']['instanceId']}
+                    log.info_5(" TaskID: " + result['json']['instanceId'])
+                    log.info_5(" Payload: " + fit_common.json.dumps(payload_data))
+                else:
+                    # if no task ID is returned put 'failed' in ID field
+                    NODE_STATUS[NODECATALOG[nodeindex]] = \
+                        {"workflow": item['workflow'],
+                         "version": item['version'],
+                         "kvm": item['kvm'],
+                         'id': "failed"}
```

Comment:
Can you log an error message here - log.error(" os install {} on node {} failed" + NODECATALOG[nodeindex] + workflow) here?  Makes it easier to see/debug a failed workflow.  thanks.